pkg/lib/hf/list.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package hf
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	kfgen "kitops/pkg/lib/kitfile/generate"
+	"kitops/pkg/output"
+)
+
+const (
+	treeURLFmt = "https://huggingface.co/api/models/%s/tree/main"
```

Comment:
Does HF allow the branch names other than `main` to be default?

Reply:
Potentially, but I've also never seen a non-main branch. I think the way to avoid this would to be allow specifying a branch, but I don't want to add a flag for that until its clear it would actually be used